crates/bevy_state/src/state/freely_mutable_state.rs

```diff
@@ -32,8 +36,24 @@ pub trait FreelyMutableState: States {
                     .in_set(StateTransitionSteps::TransitionSchedules),
             )
             .configure_sets(
-                ApplyStateTransition::<Self>::apply()
-                    .in_set(StateTransitionSteps::ManualTransitions),
+                ApplyStateTransition::<Self>::apply().in_set(StateTransitionSteps::RootTransitions),
             );
     }
 }
+
+fn apply_state_transition<S: FreelyMutableState>(
```

Comment:
This should stay public: users are explicitly encouraged to add more copies of this if they want to process state transitions at unusual times.

Reply:
This will only process transitions for FreelyMutableStates and not ComputedStates or Substates.
The latter two aren't available to users because they are closure systems.
Users should run the `StateTransition` schedule instead.

Reply:
Hmm, I can get behind that. Make sure to add this to the migration guide though.